Source/DafnyRuntime/DafnyRuntime.cs

```diff
@@ -658,7 +658,12 @@ public class Map<U, V>
 #if DAFNY_USE_SYSTEM_COLLECTIONS_IMMUTABLE
       var d = ImmutableDictionary<U, V>.Empty.ToBuilder();
 #else
-      var d = new Dictionary<U, V>();
+      // Initialize the capacity if the size of the enumerable is known
+      if (values is ICollection<Pair<U, V>> collection) {
+        var d = new Dictionary<U, V>(collection.Count);
+      } else {
+        var d = new Dictionary<U, V>();  
```

Comment:
`d` needs to be declared before the `if` statement, since it is needed downstream of the `if`.

Reply:
Thanks, I wasn't sure but I was hoping the test suite would make it obvious if I got that wrong. :) Can you point out any of the lit tests that will definitely use this code?